File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    int[] memo;
    int n;
    public int climbStairs(int n) {
        this.n = n;
        memo = new int[n+1];
        return dp(n);
    }

    public int dp(int i) {
        if (i == 1) {
            return 1;
        }
        if (i == 2) {
            return 2;
        }
        if (memo[i] == 0) {
            memo[i] = dp(i - 1) + dp(i - 2);
        }
        return memo[i];
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n==1) {
            return 1;
        }
        vector<int> memo;
        memo.push_back(1);
        memo.push_back(1);
        for (int i = 2; i <= n; i++) {
            memo.push_back(memo[i-1]+memo[i-2]);
        }
        return memo[n];
    }
};